packages/typescript-estree/src/node-utils.ts

```diff
@@ -424,7 +424,7 @@ export function findFirstMatchingAncestor(
   node: ts.Node,
   predicate: (node: ts.Node) => boolean,
 ): ts.Node | undefined {
-  while (node) {
+  while (node as ts.Node | undefined) {
```

Comment:
This will eventually cause a `no-param-reassign` error if we have that rule. Can we use another variable instead?